src/lib.rs

```diff
@@ -0,0 +1,105 @@
+//! Tasks defined rust_actix-basic.pdf
+//! Using endpoints at: https://jsonplaceholder.typicode.com/<API>
+
+#![allow(non_snake_case)] // JSON does not in general conform to rust standards, so we suppress warnings about correct case to use `serde::Deserialize` without the compiler complaining.
+
+use reqwest;
+use serde::{Serialize,Deserialize};
+use serde_json::{Value};
+
+use regex::Regex;
+
+/// Submits a GET request to a specified URL, and returns a vector of users. Expected response is in json format, and can be parsed by the `User` struct with accessible elements by key name.
+pub async fn get_users (url: &str) -> Result<Vec<User>,Box<dyn std::error::Error>> {
```

Comment:
Turn on auto-formatting in your editor ("format on save" enabled).  There are non-idiomatic formats in this file.

Reply:
thanks for the tip!